src/app/containers/ATIAnalytics/README.md

```diff
@@ -2,56 +2,130 @@
+`const eventData = event.srcElement.dataset[componentName];` 
+
+The component data object should contain a value for `creationLabel`, `child`, and optionally `adId`. These values will be used as part of the tracking URL.
+
+In the navigation example, these values were obtained by splitting the information provided via data attribute (in this example, data-attribute looks as such: `data-navigation=‘akuko_0’` - akuko is the label used for the first link in the nav, and 0 is the link index.
 
-```js
-useEffect(() => {
-    sendViewBeacon({
-      ...params,
-      element: document.querySelector('header'),
-      component: 'cookie-banner',
-    });
-}, []);
 ```
+import React, { useContext } from ‘react’;
+import { buildATIClickParams } from ‘../ATIAnalytics/params’;
+import { EventContext } from ‘#contexts/EventContext’;
+import { RequestContext } from ‘#contexts/RequestContext’;
+import { ServiceContext } from ‘#contexts/ServiceContext’;
 
-## Beacons
+const componentContainer = () => {
+	const { useClickTracker } = useContext(EventContext);
+	const eventTrackingProps = buildATIClickParams(
+		{},
+		useContext(RequestContext),
+    	useContext(ServiceContext)
+	);
 
-The `sendViewBeacon` used in the above example, is meant for impression events, such as the one sent when an element is displayed.
+	useClickTracker('[data-navigation]’, event => {
+		const eventData = event.srcElement.dataset.navigation;
+		const componentDataSplit = eventData.split(‘_’);
+		const componentData = {
+			creationLabel: componentDataSplit[0],
+	      child: componentDataSplit[1],
+	   };
+	
+		const componentInfo = getComponentInfo(event, ‘navigation’, componentData);
 
-The `sendEventBeacon` is another type of beacon, meant for events such as clicks, scrolls, etc. It sends an impression beacon, right before the event beacon.
+		// `sendEventBeacon` will go here
+	});
+}
+```
+
+All this information can then be used to put together the event beacon.
+
+```
+import React, { useContext } from ‘react’;
+import { buildATIClickParams } from ‘../ATIAnalytics/params’;
+import { EventContext } from ‘#contexts/EventContext’;
+import { RequestContext } from ‘#contexts/RequestContext’;
+import { ServiceContext } from ‘#contexts/ServiceContext’;
+import { sendEventBeacon } from ‘../ATIAnalytics/beacon’;
 
-```js
-const { useClickTracker } = useContext(EventContext);
+const componentContainer = () => {
+	const { useClickTracker } = useContext(EventContext);
+	const eventTrackingProps = buildATIClickParams(
+		{},
+		useContext(RequestContext),
+    	useContext(ServiceContext)
+	);
 
-useClickTracker('[data-consent-banner] [data-consent-accept]', e => {
-    sendEventBeacon({
-        ...params,
-        element: e.target,
-        component: 'cookie-banner',
-        type: 'click',
-        label: 'cookie-accept',
-    });
-});
+	useClickTracker('[data-navigation]’, event => {
+		const eventData = event.srcElement.dataset.navigation;
```

Comment:
Could we pass the data we need as separate attributes rather than creating and then splitting this string?

Reply:
Good point. We could do that. I assumed that we would then have to extend the functionality of `useClickTracker` to track more data attrs, but we can get that from the `event`. We can then set data attributes per information that needs to be send

Reply:
I've done a little bit of playing and we can do something along the lines of
```jsx
  useClickTracker('[data-navigation]', event => {
    const componentName = 'navigation';
    const creationLabel = event.srcElement.dataset['creationlabel'];
    const child = event.srcElement.dataset['child'];
    const componentData = {
      creationLabel,
      child,
    };

    const componentInfo = getComponentInfo({
      url: event.target.href,
      componentName,
      componentData,
    });

    sendEventBeacon({
      componentName,
      type: 'click',
      componentInfo,
      ...eventTrackingProps,
    });
  });
```

Of course, this isn't actually in scope of this ticket, but it would be good to figure this out if it's going into the Readme. We could also just put a comment in this bit of the readme saying "// get the relevant data" or something similar until we figure out our preferred approach.

Reply:
I've put this version in the readme but hopefully it's clear from the readme that this is only one way of doing it.